arangod/Aql/MaterializeExecutor.h

```diff
@@ -86,8 +86,39 @@ class MaterializerExecutorInfos {
   Collection const* _collection;
 };
 
-template<bool localDocumentId>
-class MaterializeExecutor {
+struct MaterializeExecutorBase {
+  using Infos = MaterializerExecutorInfos;
+  using Stats = MaterializeStats;
+
+  [[nodiscard]] std::tuple<ExecutorState, Stats, size_t, AqlCall> skipRowsRange(
+      AqlItemBlockInputRange& inputRange, AqlCall& call);
+
+  explicit MaterializeExecutorBase(Infos& infos);
+
+ protected:
+  transaction::Methods _trx;
+  Infos& _infos;
+};
+
+class MaterializeRocksDBExecutor : public MaterializeExecutorBase {
+ public:
+  struct Properties {
+    static constexpr bool preservesOrder = true;
+    static constexpr BlockPassthrough allowsBlockPassthrough =
+        BlockPassthrough::Enable;
+  };
+  using Fetcher = SingleRowFetcher<Properties::allowsBlockPassthrough>;
+
+  MaterializeRocksDBExecutor(Fetcher&, Infos& infos);
+
+  std::tuple<ExecutorState, MaterializeStats, AqlCall> produceRows(
+      AqlItemBlockInputRange& inputRange, OutputAqlItemRow& output);
+
+ private:
+  PhysicalCollection* _collection{};
+};
+
+class MaterializeSearchExecutor : public MaterializeExecutorBase {
```

Comment:
Why isn't it passthrough, although rocksdb passthrough?
Could you explain for me, please

Reply:
Well, you probably now better than me. Afaik it can happen that search has indexed a document that no longer exists. This is not true for rocksdb indexes (or if it is, it is a bug). Thus for the rocksdb version, we produce exactly one row for each input row. This is passthrough. In the search case, the executor acts as a filter for documents not found. Thus it will never be passthrough.

Reply:
> Afaik it can happen that search has indexed a document that no longer exists. 

We use non transactional but corresponding to inverted index rocksdb snapshot, so it shouldn't be possible

Reply:
Can you do it passthrough too? Or I can try it in the separate PR

Reply:
> Thus for the rocksdb version, we produce exactly one row for each input row.

Because same should be for search.

Reply:
> Can you do it passthrough too? Or I can try it in the separate PR
sure, please go ahead.

Reply:
> We use non transactional but corresponding to inverted index rocksdb snapshot, so it shouldn't be possible

Just to get that right, you use a different snapshot for looking up documents found via the inverted index, than the rest of the aql query?

Reply:
> Just to get that right, you use a different snapshot for looking up documents found via the inverted index, than the rest of the aql query?

Yes, even worse, can be different snapshot for different iresearch link/inverted index.

So in this materialize search executor or in view executor can be used multiple rocksdb snapshot for materialization.